AttitudeManager/Tests/AMTest.cpp

```diff
@@ -0,0 +1,295 @@
+        EXPECT_THAT(am.motorReferences_[4], ::testing::NotNull());
+
+        // Setting non-existent motors shouldn't affect yaw
+        am.outputToMotor(config::yaw, 0);
+        am.outputToMotor(config::throttle, 40);
+        am.outputToMotor(config::pitch, 30);
+        am.outputToMotor(config::roll, 20);
+
+        EXPECT_EQ(0, am.motorInstances_[0].motorInstance->percent_);
+        EXPECT_EQ(100, am.motorInstances_[1].motorInstance->percent_);
+        EXPECT_EQ(100, am.motorInstances_[2].motorInstance->percent_);
+        EXPECT_EQ(0, am.motorInstances_[3].motorInstance->percent_);
+        EXPECT_EQ(100, am.motorInstances_[4].motorInstance->percent_);
+
+        // Should set all of them
+        am.outputToMotor(config::yaw, 50);
+
+        EXPECT_EQ(50, am.motorInstances_[0].motorInstance->percent_);
+        EXPECT_EQ(50, am.motorInstances_[1].motorInstance->percent_);
+        EXPECT_EQ(50, am.motorInstances_[2].motorInstance->percent_);
+        EXPECT_EQ(50, am.motorInstances_[3].motorInstance->percent_);
+        EXPECT_EQ(50, am.motorInstances_[4].motorInstance->percent_);
+
+        // Another setting test
+        am.outputToMotor(config::yaw, 70);
+
+        EXPECT_EQ(70, am.motorInstances_[0].motorInstance->percent_);
+        EXPECT_EQ(30, am.motorInstances_[1].motorInstance->percent_);
+        EXPECT_EQ(30, am.motorInstances_[2].motorInstance->percent_);
+        EXPECT_EQ(70, am.motorInstances_[3].motorInstance->percent_);
+        EXPECT_EQ(30, am.motorInstances_[4].motorInstance->percent_);
+
+        delete control_algorithm;
+        delete[] config::motors;
+        control_algorithm = nullptr;
+        config::motors = nullptr;
+    }
+
+    // 1 motor of roll type
+    {
+        config::NUM_MOTORS = 1;
+        config::motors = new config::Motor_t[config::NUM_MOTORS];
+
+        config::motors[0] =
+            config::Motor_t{   //Roll servo motor
+                .axis = config::roll,
+                .isInverted = false,
+                .driverConstructor = config::constructObject<MotorChannel, PWMChannel,
+                                                    /*timer*/ &htim1, /*timer_channel*/ config::roll>
+            };
+
+        AM::Flightmode *control_algorithm  = config::flightmodes[0].flightmodeConstructor();
+        AM::AttitudeManager am{control_algorithm};
+
+        // Non null references
+        EXPECT_THAT(am.motorReferences_[0], ::testing::NotNull());
+        EXPECT_THAT(am.motorReferences_[1], ::testing::NotNull());
+        EXPECT_THAT(am.motorReferences_[2], ::testing::NotNull());
+        EXPECT_THAT(am.motorReferences_[3], ::testing::NotNull());
+        EXPECT_THAT(am.motorReferences_[4], ::testing::NotNull());
+
+        // Setting non-existent motors first shouldn't affect roll
+        am.outputToMotor(config::throttle, 40);
+        am.outputToMotor(config::pitch, 30);
+        am.outputToMotor(config::yaw, 20);
+        am.outputToMotor(config::roll, 10);
```

Comment:
You have verified this behaviour already in the previous test. No need to verify it again in this one.